pkg/scheduler/framework/plugins/interpodaffinity/filtering.go

```diff
@@ -119,60 +119,45 @@ func (m topologyToMatchedTermCount) updateWithTerm(node *v1.Node, t *framework.A
 
 // updateWithAffinityTerms updates the topologyToMatchedTermCount map with the specified value
 // for each affinity term if "targetPod" matches ALL terms.
-func (m topologyToMatchedTermCount) updateWithAffinityTerms(s *preFilterState, targetPod *v1.Pod, targetPodNode *v1.Node, value int64) {
-	if podMatchesAllAffinityTerms(s, targetPod) {
-		for _, t := range s.podInfo.RequiredAffinityTerms {
-			m.updateWithTerm(targetPodNode, &t, value)
+func (m topologyToMatchedTermCount) updateWithAffinityTerms(terms []framework.AffinityTerm, pod *v1.Pod, node *v1.Node, value int64) {
+	if podMatchesAllAffinityTerms(terms, pod) {
+		for _, t := range terms {
+			m.update(node, t.TopologyKey, value)
 		}
 	}
 }
 
 // updateWithAntiAffinityTerms updates the topologyToMatchedTermCount map with the specified value
 // for each anti-affinity term matched the target pod.
```

Comment:
This is not relevant here, nothing special about `nsLabels` handling with regards to it being nil or not. I added a comment in the place where we consciously pass nil because we don't need to look them up.